### DISCUSSION TITLE
Getting HLS stream from Twitch not working

### DISCUSSION THREAD

**Discussion by HellishShark** [aclap-dev/video-downloadhelper#25] [Help and Q&A] [answered]
### Your question

Hello,

in the past I always used VideoDownloadHelper to get the HLS stream from Twitch VODs, this is not working anymore. When I play the VOD, VideoDownloadHelper does not detect any media.

### Addon & CoApp version and details

version: 8.1.6.0
beta: false
build date: 2024-01-18
build options: linuxlic: false, noyt: true, browser: chrome
lang: de
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\leonh",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by jcfrog** [comment, marked as answer]
Hi,
Did you try the latest 8.2 beta? : https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions
jerome

**Comment by GrampaWildWilly** [comment]
You took the words right out of my mouth, Jérôme.  I was investigating something not even related to VDH the other day.  I downloaded an archived Twitch stream that was of duration about 90 minutes.  I am on Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0 64-bit, licensed VDH 8.2.0.18a1 beta, CoApp 2.0.14.  Works great.  I realize Twitch is as much livestreams as it is archived content, so posting a URL might not always be the most useful idea.  Still, if you're talking about archived content that you can't get, that URL would be useful.

**Comment by GrampaWildWilly** [comment]
I never visit Twitch except to investigate problems people have reported.  So I don't know much about how it works.  And the little I've looked at doesn't entice me to make it a regular stop on my own surfing tours.  In any case, I just picked this one at random: https://www.twitch.tv/clix.  As far as I can tell, it's a live stream.  Maybe it will still be going if you go there now.  Like I say, I don't really know how Twitch works.  When I told VDH to download the 1920x1080 mp4 variant it showed on its menu, it happily started recording it.  The blue dot status menu even contained an indication that it was recording a livestream.  I let it run for about a minute, then killed the download.  I had a perfectly playable (in VLC), although utterly boring, mp4 of duration 54 seconds.

Bottom line: the latest beta of the VDH browser extension as well as the most recent available CoApp work just fine on Twitch livestreams.

**Comment by HellishShark** [comment]
Hello, thanks for your suggestions, I just found out what the problem is. There is a bug related to activating the extension by click. I have VDH set to only activate when selecting the extension, VDH then activates and I have to reload the page.

With full site access, VDH detects the HLS stream flawlessly, with the setting "When you select the extension" VDH does not detect anything at all. So it does not have anything to do with beta or normal version. 

VDH worked like this in the past, with that Chrome feature on, so I assume this is a bug. Can someone add the bug label?

**Reply by paulrouget** [reply]
> VDH worked like this in the past, with that Chrome feature on, so I assume this is a bug. Can someone add the bug label?

This something we could support… but it would require reloading the page the user is current looking at, cancelling any current playback. Not great.

So I don't think there's anyway to make this work.

**Reply by HellishShark** [reply]
As far as I remember it was supported in the past. Yes it required reloading of the page, but that's not a big problem in my opinion, especially when someone is only downloading VODs once a while like me. Twitch immediately restarts the playback when reloading so VDH should have something to detect.

**Reply by paulrouget** [reply]
I think it should "kind of work" if you reload the page and quickly click on the button and wait.

But maybe we can add an option for this type of scenario. I'll look into it.
